go.mod

```diff
@@ -1,14 +1,16 @@
 module github.com/microsoft/retina
 
-go 1.22.3
+go 1.22.5
```

Comment:
1.22.5 is required for sigs.k8s.io/cloud-provider-azure/pkg/azclient

Reply:
Also, CVEs are reported in 1.22.3 https://github.com/Azure/azure-storage-azcopy/issues/2704
```

┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                   Title                    │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24789 │ UNKNOWN  │ fixed  │ 1.22.3            │ 1.21.11, 1.22.4 │ https://avd.aquasec.com/nvd/cve-2024-24789 │
│         ├────────────────┤          │        │                   │                 ├────────────────────────────────────────────┤
│         │ CVE-2024-24790 │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24790 │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴────────────────────────────────────────────┘
```
and in 1.22.4 https://github.com/Azure/azure-storage-azcopy/issues/2744
```
┌─────────┬────────────────┬──────────┬────────┬───────────────────┬─────────────────┬──────────────────────────────────────────────────────────┐
│ Library │ Vulnerability  │ Severity │ Status │ Installed Version │  Fixed Version  │                          Title                           │
├─────────┼────────────────┼──────────┼────────┼───────────────────┼─────────────────┼──────────────────────────────────────────────────────────┤
│ stdlib  │ CVE-2024-24791 │ MEDIUM   │ fixed  │ 1.22.4            │ 1.21.12, 1.22.5 │ net/http: Denial of service due to improper 100-continue │
│         │                │          │        │                   │                 │ handling in net/http                                     │
│         │                │          │        │                   │                 │ https://avd.aquasec.com/nvd/cve-2024-24791               │
└─────────┴────────────────┴──────────┴────────┴───────────────────┴─────────────────┴──────────────────────────────────────────────────────────┘
```

Reply:
Version bump is appreciated, but the `go` directive in `go.mod` is not a security feature--it's to declare the language features needed to build the module. This should probably be `toolchain`. See: https://go.dev/doc/toolchain and https://go.dev/blog/toolchain

Reply:
I bumped azclients and 1.23.5 is not required. So I just reverted bumping go version.
Also I read through the documentations, and indeed it's an toolchain change if we still need 1.23.5